### ISSUE TITLE
NewType support

### ISSUE THREAD

**Issue by m09** [BrianPugh/cyclopts#262]
[`NewType`](https://docs.python.org/3/library/typing.html#newtype)'s magic seems to confuse `cyclopts` at the moment. On my Python 3.10, `cyclopts` expects two values when I annotate a command argument with a `str` derived instance of `NewType`. 

It is possible to detect `NewType`s in the following way on my Python 3.10, although methods may vary between versions since `NewType` changed in 3.10 & 3.11:

```python
>>> A = NewType("A", str)
>>> isinstance(A, NewType)
True
>>> A.__supertype__
<class 'str'>
```

**Comment by BrianPugh** [comment]
Thanks for opening the issue! I haven't ever actually seen `NewType` before, but I'm sure we can probably add support. I'll investigate soon, thanks!

**Comment by m09** [comment]
Thanks a lot Brian!

**Comment by m09** [comment]
I had time to test things a bit more and it seems that I proceeded too hastily yesterday during my testing session: I got lists working but only because I used an even number of items during my tests. It seems that an odd number of items in the list will make `cyclopts` fail.

MWE:

```python
from typing import NewType

from cyclopts import App

app = App()
A = NewType("A", str)


@app.default()
def main(items: list[A]) -> None:
    print(items)
```
Output:

```
❯ mwe a                  
╭─ Error ────────────────────────────────────────────────────────────────────────────────────╮
│ Parameter "--items" requires 2 arguments. Only got 1.                                      │
╰────────────────────────────────────────────────────────────────────────────────────────────╯

❯ mwe a b
['a', 'b']

❯ mwe a b c
╭─ Error ────────────────────────────────────────────────────────────────────────────────────╮
│ Parameter "--items" requires 2 arguments. Only got 1.                                      │
╰────────────────────────────────────────────────────────────────────────────────────────────╯

❯ mwe a b c d
['a', 'b', 'c', 'd']

❯ 
```

**Comment by BrianPugh** [comment]
seems like this current issue is specific to some versions of python; investigating.

**Comment by BrianPugh** [comment]
this should now be fixed in v3.1.2!

**Comment by m09** [comment]
Thanks a lot Brian :+1:
